docs/packages.md

```diff
@@ -0,0 +1,124 @@
+
+The templates are very simple; each is just a project file and a
+single source file. The project is just a console application with a
+dependency on Google.Cloud.Functions.Invoker. If you're happy
+creating the project yourself, you definitely don't *need* to use
+the templates. You can create a regular console application, replace
+the `Program` file with a function, add the dependency, and
+you're away.
+
+## Google.Cloud.Functions.Invoker.Testing
+
+The testing package makes it easier to write tests which start an
+in-memory ASP.NET Core server to respond to requests. It's not
+generally needed to write unit tests for functions, and it's not
+*required* even to write integration tests - it's just convenient.
+
+See the [testing documentation](testing.md) for more details and
+links to sample code.
+
+## Google.Cloud.Functions.Framework
+
+This package is the core of the Functions Framework for .NET. It
+contains:
+
+- The interfaces that functions are expected to implement
+  (`IHttpFunction`, `ICloudEventFunction` etc)
+- Adapter classes to make it easy to wrap any function other tha
+  `IHttpFunction` *as* an `IHttpFunction`.
+- Legacy event types (e.g. `StorageObject`, `PubSubMessage` and
+  `FirestoreEvent`)
+
+The framework depends on ASP.NET Core, and also the
+[CloudNative.CloudEvents
+package](https://github.com/cloudevents/sdk-csharp) for
+[CNCF Cloud Events](https://cloudevents.io/).
+
+Cloud Functions can depend on just the types within this package in
+order to be portable however they're hosted.
+
+## Google.Cloud.Functions.Invoker
+
+The invoker package is expected to be used as a dependency from a
+*project*, but not usually from *source code* itself. For example,
+you won't see any `using` directives for
+the `Google.Cloud.Functions.Invoker` namespace in the samples or
+templates, even though the projects refer to the package.
+
+The purpose of the package is to allow you to get up and running
+with absolutely minimal code. When a function package depends on the
+invoker package, an entry point (a class with a `Main` method) is
+generated automatically via MSBuild magic when the function package
+is built. That entry point calls into `EntryPoint.StartAsync` method
+within the invoker package to start the web server.
+
+(Note: the entry point generation can be disabled by setting the
+`AutoGenerateEntryPoint` MSBuild property to any value other than
+`true`, which is the default.)
+
+The main roles of the invoker are:
+
+- Handle the environment variables and any command line arguments to
+  work out which function to invoke, what port to listen on etc.
+- Set up ASP.NET Core dependency injection, including any adapters
+  required to handle your function.
+- Set up logging in a manner appropriate for your environment.
+- Run the ASP.NET Core server, with a pipeline to call the function.
+
+## Creating a function server without the invoker package
+
+Nothing ties Cloud Run (or Cloud Functions) to the invoker
```

Comment:
This comes back to the general discussion, but this one is slightly different, because it *is* specific to Run and Functions.

Maybe a later product comes along - perhaps on Azure or AWS - that *does* tie in to the invoker. We can't speak generally about what all function-like products will do - but we *can* give specific information about Cloud Run and Cloud Functions.

Of course, we could not give this information at all - but then that can leave the users in doubt about whether or not they have to use our packages.